### PR TITLE
TPC QC: make IDC plots (v. 1)

### PR DESCRIPTION
**Pull request by tklemenz**

### CODE REVIEW

**Comment by tklemenz** [comment]
This is the first version of the IDCs visualization for QC.

Fourier coefficients are not yet included (timestamp -1 for getting the data from the CCDB is not working for them for now) and there is a fit missing to the radial IDC0 distribution. 
This will come in a later update.

**Comment by tklemenz** [comment]
Clang format gives an error at the moment, that is why the check is failing.

`$ clang-format-6.0 -style=file -i IDCs.cxx`
`YAML:52:19: error: unknown key 'DeriveLineEnding'`
`DeriveLineEnding: true`
`Error reading /home/tklemenz/AliSoftware/QualityControl/.clang-format: Invalid argument`

I will investigate.

**Comment by tklemenz** [comment]
The issue with clang-format is fixed and the Fourier Coefficients are now included.

File: Detectors/TPC/calibration/include/TPCCalibration/IDCCCDBHelper.h
Comment (wiechula) [inline Detectors/TPC/calibration/include/TPCCalibration/IDCCCDBHelper.h]: Why do you need these functions twice, once in the private and then wrapped here in the public part? Can't you simply move the definitions from the private part here?

File: Detectors/TPC/calibration/include/TPCCalibration/IDCDrawHelper.h
Comment (wiechula) [inline Detectors/TPC/calibration/include/TPCCalibration/IDCDrawHelper.h:61]: If you plan to use `TH2Poly` in QC one has to be very careful, I think. These will be huge ...

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx]:
```suggestion
  auto hAside2D = IDCDrawHelper::drawSide(drawFun, Side::A, zAxisTitle);
  auto hCside2D = IDCDrawHelper::drawSide(drawFun, Side::C, zAxisTitle);
```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx]:
```suggestion
  auto hAside1D = IDCDrawHelper::drawSide(drawFun, fmt::format("{}", type.data()).data(), Side::A, nbins1D, xMin1D, xMax1D);
  auto hCside1D = IDCDrawHelper::drawSide(drawFun, fmt::format("{}", type.data()).data(), Side::C, nbins1D, xMin1D, xMax1D);
```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx:174]:
```suggestion
  hAside2D->SetBit(TObject::kCanDelete);
  hCside2D->SetBit(TObject::kCanDelete);
  hAside1D->SetBit(TObject::kCanDelete);
  hCside1D->SetBit(TObject::kCanDelete);

```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx]:
```suggestion
```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx]: Our default unit is `cm`, please adapt the code.

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx]:
```suggestion
  auto hCside2D = new TH2F("h_IDC0_radialProfile_Cside", "IDC0: Radial profile (C-Side)", radialBinning.size()-1, radialBinning.data(), nbinsY, yMin, yMax);
```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx:235]:
```suggestion

  hAside2D->SetBit(TObject::kCanDelete);
  hCside2D->SetBit(TObject::kCanDelete);
```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx:316]:
```suggestion
  canv->SetBit(TObject::kCanDelete);
```

File: Detectors/TPC/calibration/src/IDCDrawHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCDrawHelper.cxx]: readability braces
```suggestion
  std::string sideName = (side == Side::A) ? "A" : "C";
```

File: Detectors/TPC/calibration/include/TPCCalibration/IDCDrawHelper.h
Comment (wiechula) [inline Detectors/TPC/calibration/include/TPCCalibration/IDCDrawHelper.h]:
```suggestion
  static void drawRadialProfile(const IDCDraw& idc, TH2F& hist, const o2::tpc::Side side, const int nbinsY, const float yMin, const float yMax);
```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx]:
```suggestion
  IDCDrawHelper::drawRadialProfile(drawFun, *hAside2D, Side::A, nbinsY, yMin, yMax);
  IDCDrawHelper::drawRadialProfile(drawFun, *hCside2D, Side::C, nbinsY, yMin, yMax);
```

File: Detectors/TPC/calibration/src/IDCDrawHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCDrawHelper.cxx]:
```suggestion
void o2::tpc::IDCDrawHelper::drawRadialProfile(const IDCDraw& idc, TH2F& hist, const o2::tpc::Side side)
```

File: Detectors/TPC/calibration/src/IDCDrawHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCDrawHelper.cxx]:
```suggestion
void o2::tpc::IDCDrawHelper::drawIDCZeroStackCanvas(const IDCDraw& idc, const o2::tpc::Side side, const std::string_view type, int nbins1D, float xMin1D, float xMax1D, TCanvas& outputCanvas, int integrationInterval)
```
And required changes below.

File: Detectors/TPC/calibration/include/TPCCalibration/IDCDrawHelper.h
Comment (wiechula) [inline Detectors/TPC/calibration/include/TPCCalibration/IDCDrawHelper.h]:
```suggestion
  static void drawIDCZeroStackCanvas(const IDCDraw& idc, const o2::tpc::Side side, const std::string_view type, const int nbins1D, const float xMin1D, const float xMax1D, TCanvas& outputCanvas, int integrationInterval);
```

File: Detectors/TPC/calibration/src/IDCDrawHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCDrawHelper.cxx:85]: call this function in the one above.

File: Detectors/TPC/calibration/src/IDCDrawHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCDrawHelper.cxx]:
```suggestion
  const auto padNum = Mapper::getGlobalPadNumber(irow, ipad, region);
  const float padX = mapper.padCentre(padNum).x();
  hist.Fill(padX, idc.getIDC(sector, region, irow, padTmp));
```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx]: I added a function for the binning to the Painter: https://github.com/AliceO2Group/AliceO2/pull/8772
Once it is merged you can use it:
```suggestion
  const auto radialBinning = painter::getRowBinningCM();
  
  auto hAside2D = new TH2F("h_IDC0_radialProfile_Aside", "IDC0: Radial profile (A-Side)", radialBinning.size()-1, radialBinning.data(), nbinsY, yMin, yMax);
```

File: Detectors/TPC/calibration/src/IDCDrawHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCDrawHelper.cxx]:
```suggestion
const mapper& = Mapper::instance();
```

File: Detectors/TPC/calibration/src/IDCDrawHelper.cxx
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCDrawHelper.cxx]:
```suggestion
          if (region < 4) {
            hIROC->Fill(idc.getIDC(sector, region, irow, padTmp));
          } else if (region < 6) {
            hOROC1->Fill(idc.getIDC(sector, region, irow, padTmp));
          } else if (region < 8) {
            hOROC2->Fill(idc.getIDC(sector, region, irow, padTmp));
          } else {
            hOROC3->Fill(idc.getIDC(sector, region, irow, padTmp));
         }
```

**Review by wiechula — CHANGES_REQUESTED** [review]

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx
Comment (tklemenz) [reply Detectors/TPC/calibration/src/IDCCCDBHelper.cxx:316]: I think this should stay as it is.

**Comment by tklemenz** [comment]
I implemented everything except the binning, where I wait for the PR in the painter to be merged. 
@wiechula  Also please check the one comment above.

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx
Comment (wiechula) [reply Detectors/TPC/calibration/src/IDCCCDBHelper.cxx:316]: Yes, sure. My bad!

**Review by wiechula — CHANGES_REQUESTED** [review]
https://github.com/AliceO2Group/AliceO2/pull/8772 was merged, please change your code as suggested.

**Review by wiechula — APPROVED** [review]

**Comment by tklemenz** [comment]
Should I leave WIP until the PR it depends on is merged?

**Comment by wiechula** [comment]
Isn't it already merged?

**Comment by tklemenz** [comment]
Ah, my bad. I looked at the wrong PR. So I remove it.

**Comment by wiechula** [comment]
@shahor02 , @davidrohr , can this be merged?

**Comment by wiechula** [comment]
can this be merged?

**Pull request merged**
